packages/block-library/src/navigation/editor.scss

```diff
@@ -237,6 +237,12 @@ $color-control-label-height: 20px;
 	position: relative;
 	z-index: 1;
 
+	// If an ancestor has a text-decoration property applied, it is inherited regardless of
+	// the specificity of a child element. Only pulling the child out of the flow fixes it.
+	// See also https://www.w3.org/TR/CSS21/text.html#propdef-text-decoration.
+	float: left;
```

Comment:
Thanks for investigating! TIL. 

It looks like `display: inline-block;` also works, but I don't really have an opinion on whether that's any better or worse than a float.